cmd/gardenlet/app/app.go

```diff
@@ -284,19 +287,27 @@ func (g *garden) Start(ctx context.Context) error {
 				},
 			}
 
-			ns := gardenerutils.ComputeGardenNamespace(g.config.SeedConfig.SeedTemplate.Name)
+			seedNamespace := gardenerutils.ComputeGardenNamespace(g.config.SeedConfig.SeedTemplate.Name)
 			return kubernetes.AggregatorCacheFunc(
 				kubernetes.NewRuntimeCache,
 				map[client.Object]cache.NewCacheFunc{
 					// Gardenlet should watch secrets only in the seed namespace of the seed it is responsible for. We
 					// don't use any selector mechanism here since we want to still fall back to reading secrets with
 					// the API reader (i.e., not from cache) in case the respective secret is not found in the cache.
 					&corev1.Secret{}: func(c *rest.Config, o cache.Options) (cache.Cache, error) {
-						o.Namespaces = []string{ns, ns}
+						// cache.New only creates a multiNamespacedCache if cache.Options.Namespaces has more than one namespace
+						// (https://github.com/kubernetes-sigs/controller-runtime/blob/116a1b831fffe7ccc3c8145306c3e1a3b1b14ffa/pkg/cache/cache.go#L202-L204).
+						// MultiNamespacedCache can only be possible when there are more than one namespace
+						// (https://github.com/kubernetes-sigs/controller-runtime/blob/116a1b831fffe7ccc3c8145306c3e1a3b1b14ffa/pkg/cache/multi_namespace_cache.go#L53-L55)
```

Comment:
Can you also add an explanation why we need a multi namespaced cache here?

Reply:
Above line https://github.com/gardener/gardener/blob/5c71cb98dcf958a31b583fec75d22e16b0b04322/cmd/gardenlet/app/app.go#L294-L296 already explains it right?